src/Meditation.PatchingService/Services/PatchListProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Meditation.PatchingService.Models;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Runtime.Loader;
+using Meditation.PatchLibrary;
+
+namespace Meditation.PatchingService.Services
+{
+    internal class PatchListProvider : IPatchListProvider
+    {
+        private readonly IPatchStorage _patchStorage;
+        private readonly AssemblyLoadContext _assemblyLoadContext;
+        private ImmutableArray<PatchInfo> _patches;
+        private bool _isReloadRequested;
+
+        public PatchListProvider(IPatchStorage patchStorage)
+        {
+            _patchStorage = patchStorage;
+            _assemblyLoadContext = new AssemblyLoadContext(name: nameof(PatchListProvider));
+            _isReloadRequested = true;
+        }
+
+        public void Reload()
+        {
+            _isReloadRequested = true;
+        }
+
+        public ImmutableArray<PatchInfo> GetAllPatches()
+        {
+            if (!_isReloadRequested)
+                return _patches;
+                    
+            var files = Directory.GetFiles(_patchStorage.GetRootFolderForPatches())
+                .Where(f => string.Equals(Path.GetExtension(f), ".dll", StringComparison.InvariantCultureIgnoreCase));
+
+            var patchesBuilder = new List<PatchInfo>();
+            foreach (var file in files)
```

Comment:
It will be better to filter files by name before trying to load (loading of all patches in folder looks not correct and can produce exceptions)

upd: we discussed, the patch loading mechanism is unfinished and will be reworked

Reply:
The patching mechanism will be reimplemented. We need to have an approach as described here: #38